helpers/craigslist_scraper.rb

```diff
@@ -0,0 +1,77 @@
+require 'mechanize'
+require_relative 'posting'
+
+class CraigslistScraper
+  attr_reader :search_page, :scraper, :results_page, :postings
+
+  def initialize(min_price, max_price, keywords)
+    @scraper = Mechanize.new
+    @scraper.history_added = Proc.new { sleep 0.5 }
+    @search_page = scraper.get('http://sfbay.craigslist.org/search/apa')
+    form = search_page.forms[1]
+    form["minAsk"] = min_price
+    form["maxAsk"] = max_price
+    form["query"] = keywords
+    @results_page = scraper.submit(form)
+    @results_nodes = @results_page.search("div.content p.row")
+    @postings = get_postings
+    append_emails
+  end
+
+  def render
+    html = ""
+    @postings.each do |post|
+      html += "<tr>"
+      html += "<td>#{post.name}</td>"
+      html += "<td>#{post.price}</td>"
+      html += "<td>#{post.location}</td>"
+      html += "<td><a href='#{post.url}'>link</a></td>"
+      if post.email
+        html += "<td>#{post.email}</td>"
+      end
```

Comment:
Interesting approach to this; I just plugged mine directly in. Erik said on mine maybe it could be a candidate for a partial.